RecoTracker/FinalTrackSelectors/plugins/TfGraphDefProducer.cc

```diff
@@ -0,0 +1,66 @@
+// -*- C++ -*-
+//
+// Package:    test/TFGraphDefProducer
+// Class:      TFGraphDefProducer
+//
+/**\class TFGraphDefProducer
+ Description: Produces TfGraphRecord into the event containing a tensorflow GraphDef object that can be used for running inference on a pretrained network
+*/
+//
+// Original Author:  Joona Havukainen
+//         Created:  Fri, 24 Jul 2020 08:04:00 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "TrackingTools/Records/interface/TfGraphRecord.h"
+#include "RecoTracker/FinalTrackSelectors/interface/TfGraphDefWrapper.h"
+
+// class declaration
+
+class TfGraphDefProducer : public edm::ESProducer {
```

Comment:
Just to note that if this class was made an ESSource, an `EmptyESSource` for the `TfGraphRecord` would not be needed. (depends mostly if other ESProducers producing products to `TfGraphRecord` that would not consume the product of `TfGraphDefProducer` are foreseen)

Reply:
Given that the ESProducer itself is generic (even if the Session is made part of the ESProduct) I can easily imagine other users in the future, in which case it would be better to stay using `EmptyESSource` and not make this module ESSource.